indico/modules/auth/forms.py

```diff
@@ -46,22 +47,33 @@ def _check_not_blacklisted(form, field):
             raise ValidationError(error_msg)
 
 
+def _check_use_own_email_as_username(form, field):
+    if validate_email(field.data, check_dns=False):
+        if field.data not in form.user.all_emails:
+            raise ValidationError(_('Only emails addresses linked to your account can be used as a username.'))
+
+
```

Comment:
@ThiefMaster, @OmeGak, About this validation `_check_use_own_email_as_username` what we actually discussed was to prevent users from having characters such as '@' in their username and to do the validation on new usernames. 

But IMO this check would not be sufficient because it would prevent usernames like 'John@' which seems like a reasonable username. So what I did was to prevent users from using an email as an identifier that isn't their own email. @OmeGak raised a point yesterday that if a user uses an email as identifier then they can later remove that email from their emails and would then have an identifier that isn't their own email. 

If either of you have a strong opinion about definitely not allowing emails as usernames then I can remove this validation and just ensure no email is used as username. (For existing usernames to remain valid there'll be extra checks for if old_data != new_data)

Reply:
I don't think `John@` needs to be allowed as a username.

TBH I'd even be OK with just allowing alphanumeric usernames at some point, but there's no need for this right now...

Reply:
Should we just disallow emails as the username in general, and make the username an optional field? That way logging in with email/password would always work, and if a user wants to set a username they can...

Reply:
It should be ok not allowing emails as usernames. But isn't there a constraints on the identifier for it to always have a value? If so then they can continue to have usernames but login with email address if they want

Reply:
Yes, it would need a change to the DB model to allow the Identity to have a null identifier when it's using the indico provider.